src/utilities/modules/temperatureSensorHelpers.js

```diff
@@ -370,87 +383,167 @@ function addHeadAndTrailingLogToBreach({ result, sensor, database }) {
   return { ...result, numberOfBreachHedOrTailLogs: headOrTrailLogs.length };
 }
 
-function doFullAggregation({ result, sensor, database }) {
-  const { sensorLogs } = sensor;
-
-  let lastFullAggregation = null;
-  let aggregateFrom = sensorLogs
-    .filtered('aggregation == "aggregate" || aggregation == "breachAggregate"')
-    .max('timestamp');
-  const allPreAggregateLogs = sensorLogs.filtered('aggregation == "preAggregate"');
-
-  if (!aggregateFrom) aggregateFrom = allPreAggregateLogs.min('timestamp');
-  else {
-    lastFullAggregation = aggregateFrom;
-    aggregateFrom = new Date(aggregateFrom.getTime() + 1);
-  }
-
-  if (!aggregateFrom) {
-    return { ...result, noRecordsToFullAggregate: true };
-  }
-
-  const allPreAggregateMaxDate = allPreAggregateLogs.max('timestamp');
-  const logsToAdd = [];
-
-  const halfFullAggregateInterval = Math.floor(fullAggregateInterval / 2);
-  let aggregateTo = new Date(aggregateFrom.getTime() + fullAggregateInterval);
-  let aggregateMiddlePoint = new Date(aggregateFrom.getTime() + halfFullAggregateInterval);
-  let lastAggregationTo = null;
-
-  while (allPreAggregateMaxDate >= aggregateTo) {
-    const thisPeriodLogs = allPreAggregateLogs.filtered(
-      'timestamp >= $0 && timestamp < $1',
-      aggregateFrom,
-      aggregateTo
-    );
-
-    logsToAdd.push({
+function linkSensorLogToItemBatches({ sensorLog, database }) {
+  const { location } = sensorLog;
+  if (!location) return;
+  const itemBatches = location.getItemBatchesWithQuantity(database);
+  if (!itemBatches || itemBatches.length === 0) return;
+  database.update('SensorLog', {
+    ...sensorLog,
+    itemBatches,
+  });
+  itemBatches.forEach(itemBatch =>
```

Comment:
also add sensorLog to itemBatches please

Reply:
da613bfbbc6d41ab60e2c213d675915ab62295fe


Added a helper to itemBatches to be consistent with other realm models?